src/resource/crc32.rs

```diff
@@ -23,34 +31,37 @@ use crate::{ArklibError, Result};
     Deserialize,
     Serialize,
 )]
-pub struct ResourceId {
+pub struct ResourceIdCrc32 {
     pub data_size: u64,
-    pub crc32: u32,
+    pub hash: u32,
 }
 
-impl Display for ResourceId {
+impl Display for ResourceIdCrc32 {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "{}-{}", self.data_size, self.crc32)
+        write!(f, "{}-{}", self.data_size, self.hash)
     }
 }
 
-impl FromStr for ResourceId {
+impl FromStr for ResourceIdCrc32 {
     type Err = ArklibError;
 
     fn from_str(s: &str) -> Result<Self> {
         let (l, r) = s.split_once('-').ok_or(ArklibError::Parse)?;
         let data_size: u64 = l.parse().map_err(|_| ArklibError::Parse)?;
-        let crc32: u32 = r.parse().map_err(|_| ArklibError::Parse)?;
+        let hash: u32 = r.parse().map_err(|_| ArklibError::Parse)?;
 
-        Ok(ResourceId { data_size, crc32 })
+        Ok(ResourceIdCrc32 { data_size, hash })
     }
 }
 
-impl ResourceId {
-    pub fn compute<P: AsRef<Path>>(
-        data_size: u64,
-        file_path: P,
-    ) -> Result<Self> {
+impl ResourceIdTrait<'_> for ResourceIdCrc32 {
+    type HashType = u32;
+
+    fn get_hash(&self) -> Self::HashType {
+        self.hash
+    }
```

Comment:
Not sure that we need to expose this. Probably ids should be "black boxes", i.e. users should only use them (via `Eq`, `PartialEq`, `Ord`, `Display` traits) without going into the details of how the ids are constructed.

Reply:
Currently, `ResourceId` has `data_size` and `hash` as public fields and I see them used in many places all over the codebase (especially in unit tests). 
I thought if the hash is so frequently used, might be a good idea to add a method to get it. 

What do you think?

Reply:
Let's consider this by use-cases:
1. Unit tests should be able to access internal fields, but they should not be exposed to users/apps.
2. Users/apps could use `data_size` but if they need it, they would need it with other hash functions, too. It's better to provide such a field in some other way, generically for any hash function. Since we just pass this data to crc32 generation function, we could  save it somewhere around. I don't think that users really need to do something with `hash` value, only with the id as a whole.

Reply:
> Unit tests should be able to access internal fields

In the current setup, implementers of `ResourceIdTrait` have the freedom to decide whether to expose `data_size` and `hash` or not. It may be best not to enforce methods like `get_hash` in the trait itself.

Same goes for any methods related to `data_size`, considering that this field might not be applicable in all implementations of `ResourceIdTrait`, especially in cryptographic hash function implementations.

Reply:
We didn't include `data_size` in the draft for `BLAKE3` implementation. Here https://github.com/ARK-Builders/arklib/pull/73#discussion_r1468890690